libs/beacon/src/beacon_manager.cpp

```diff
@@ -169,13 +158,7 @@ std::unordered_set<BeaconManager::MuddleAddress> BeaconManager::ComputeComplaint
 bool BeaconManager::VerifyComplaintAnswer(MuddleAddress const &from, ComplaintAnswer const &answer)
 {
   CabinetIndex from_index = identity_to_index_[from];
-  if (identity_to_index_.find(answer.first) == identity_to_index_.end())
-  {
-    FETCH_LOG_WARN(LOGGING_NAME, "Node ", cabinet_index_,
-                   " received complaint answer with unknown reporter index");
-    return true;
-  }
-
+  assert(identity_to_index_.find(answer.first) != identity_to_index_.end());
```

Comment:
So, just to check, assertions should be used for something that should never be able to happen. There is no way a node can submit information to get past this point is there?

Reply:
It is checked in beacon_setup_service that answer.first is in identity_to_index so the check here was redundant. I put the assert there to replace it for exactly the reason you said, it should not happen.